### PR TITLE
Remove UI states from test certs (EXPOSUREAPP-8885, 8886)

### PR DESCRIPTION
**Pull request by SamuraiKek**
- Scan every type of certificate https://dgc.a-sit.at/ehn/. on WRU.
- Travel to the future one year - 1 day from today and check the expiry status of all certs.
- Travel to the future one year + 1 day from today and check the status of all certs.
- Remove the recovery and vaccination certs and test the app with only the test certs inside.

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/items/TestCertificateCard.kt [hunk 1]
```diff
@@ -41,18 +41,13 @@ class TestCertificateCard(parent: ViewGroup) :
...
-            else -> PersonColorShade.COLOR_INVALID
-        }
+        val color = curItem.colorShade
+        certificateIcon.setImageResource(R.drawable.ic_test_certificate)
 
-        when {
-            curItem.certificate.isValid -> R.drawable.ic_test_certificate
-            else -> R.drawable.ic_certificate_invalid
-        }.also { certificateIcon.setImageResource(it) }
+        certificateIcon.setImageResource(R.drawable.ic_test_certificate)
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/items/TestCertificateCard.kt:50]: repeated line ?

Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/items/TestCertificateCard.kt:50]: Oops, looks like it.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/PersonDetailsViewModel.kt [hunk 2]
```diff
@@ -88,11 +88,10 @@ class PersonDetailsViewModel @AssistedInject constructor(
     private suspend fun assembleList(personCertificates: PersonCertificates, isLoading: Boolean) =
         mutableListOf<CertificateItem>().apply {
             val priorityCertificate = personCertificates.highestPriorityCertificate
-
-            when {
-                priorityCertificate.isValid -> colorShade
-                else -> PersonColorShade.COLOR_INVALID
-            }.also { colorShadeData.postValue(it) }
+            val colorState = if (priorityCertificate.isValid || priorityCertificate is TestCertificate) {
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/PersonDetailsViewModel.kt:91]: Invalid Test Certificate should be displayed as invalid , right? we should just exclude expiring and expiring_soon states of the test certificate

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/PersonDetailsViewModel.kt:91]: in 8886 , the tester mentioned that it is not expected for expired TC to show as `invalid` ,but it does not mean we should not display invalid TC as invalid

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/items/PersonCertificateCard.kt [hunk 3]
```diff
@@ -30,10 +31,9 @@ class PersonCertificateCard(parent: ViewGroup) :
     ) -> Unit = { item, payloads ->
         val curItem = payloads.filterIsInstance<Item>().singleOrNull() ?: item
 
-        val color = when {
-            curItem.certificate.isValid -> curItem.colorShade
-            else -> PersonColorShade.COLOR_INVALID
-        }
+        val color = if (curItem.certificate.isValid || curItem.certificate is TestCertificate) {
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/items/PersonCertificateCard.kt:34]: same here

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/PersonDetailsViewModel.kt [hunk 4]
```diff
@@ -88,11 +88,10 @@ class PersonDetailsViewModel @AssistedInject constructor(
...
+                colorShade
+            } else PersonColorShade.COLOR_INVALID
+            colorShadeData.postValue(colorState)
```
[the remark this answers is not in the document]
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/PersonDetailsViewModel.kt:94]: Sure, I can change it back. It felt a bit weird to me to have a when statement that can be replaced by an if statement.

**Review by ralfgehrer — APPROVED** [review]
Did not test on device. Coding lgtm.

**Review by BMItr — APPROVED** [review]
👌

**Pull request merged**